tfhe/src/core_crypto/fft_impl/f128/crypto/tests.rs

```diff
@@ -0,0 +1,221 @@
+        for a_i in accumulator_scalar[0..half_box_size].iter_mut() {
+            *a_i = (*a_i).wrapping_neg();
+        }
+
+        // Rotate the accumulator
+        accumulator_scalar.rotate_left(half_box_size);
+
+        let accumulator_plaintext = PlaintextList::from_container(accumulator_scalar);
+
+        allocate_and_trivially_encrypt_new_glwe_ciphertext(glwe_size, &accumulator_plaintext)
+    }
+
+    let f = |x: Scalar| Scalar::TWO * x;
+    let accumulator: GlweCiphertextOwned<Scalar> = generate_accumulator(
+        polynomial_size,
+        glwe_dimension.to_glwe_size(),
+        message_modulus.cast_into(),
+        delta,
+        f,
+    );
+
+    // Allocate the LweCiphertext to store the result of the PBS
+    let mut pbs_multiplication_ct: LweCiphertext<Vec<Scalar>> =
+        LweCiphertext::new(Scalar::ZERO, big_lwe_sk.lwe_dimension().to_lwe_size());
+    println!("Computing PBS...");
+
+    fourier_bsk.bootstrap(
+        &mut pbs_multiplication_ct,
+        &lwe_ciphertext_in,
+        &accumulator,
+        &fft,
+        PodStack::new(&mut GlobalPodBuffer::new(
+            K::bootstrap_scratch(
+                std_bootstrapping_key.glwe_size(),
+                std_bootstrapping_key.polynomial_size(),
+                &fft,
+            )
+            .unwrap(),
+        )),
+    );
+
+    // Decrypt the PBS multiplication result
+    let pbs_multipliation_plaintext: Plaintext<Scalar> =
+        decrypt_lwe_ciphertext(&big_lwe_sk, &pbs_multiplication_ct);
+
+    // Create a SignedDecomposer to perform the rounding of the decrypted plaintext
+    // We pass a DecompositionBaseLog of 5 and a DecompositionLevelCount of 1 indicating we want to
+    // round the 5 MSB, 1 bit of padding plus our 4 bits of message
+    let signed_decomposer =
+        SignedDecomposer::new(DecompositionBaseLog(5), DecompositionLevelCount(1));
+
+    // Round and remove our encoding
+    let pbs_multiplication_result: Scalar =
+        signed_decomposer.closest_representable(pbs_multipliation_plaintext.0) / delta;
+
+    println!("Checking result...");
+    assert_eq!(f(input_message), pbs_multiplication_result);
+    println!(
+        "Mulitplication via PBS result is correct! Expected 6, got {pbs_multiplication_result}"
+    );
+}
+
+#[test]
+fn test_bootstrap_u128() {
+    test_bootstrap_generic::<u128, Fourier128LweBootstrapKeyOwned>(
+        StandardDev(sqr(0.000007069849454709433)),
+        StandardDev(sqr(0.00000000000000029403601535432533)),
+    );
+}
+
+#[test]
+fn test_bootstrap_u64() {
+    test_bootstrap_generic::<u64, Fourier128LweBootstrapKeyOwned>(
+        StandardDev(sqr(0.000007069849454709433)),
+        StandardDev(sqr(0.00000000000000029403601535432533)),
+    );
+}
+
+#[test]
+fn test_bootstrap_u32() {
+    test_bootstrap_generic::<u32, Fourier128LweBootstrapKeyOwned>(
+        StandardDev(sqr(0.000007069849454709433)),
+        StandardDev(sqr(0.00000000000000029403601535432533)),
+    );
+}
```

Comment:
can we have the tests for the u64 f64 using the same interface as well as the split version of the PBS f128_u128 ?

Reply:
the split version is tested here, since the f128 bootstrap uses the split version when Scalar == u128

Reply:
i can add the test to the f64 code

Reply:
ah perfect! I did not realize it was the case

Reply:
yes please for f64 :)